store/sqlstore/channel_store.go

```diff
@@ -3467,159 +3443,122 @@ func (s SqlChannelStore) buildFulltextClauseX(term string, searchColumns ...stri
-                        cc.Id
-                    HAVING
-                        %s
-                    LIMIT
-                        ` + strconv.Itoa(model.ChannelSearchDefaultLimit) + `
-                )`
-	} else {
-		baseLikeClause = "GROUP_CONCAT(u.Username SEPARATOR ', ') LIKE %s"
-		query = `
-            SELECT
-                cc.*
-            FROM (
-                SELECT
-                    c.*
-                FROM
-                    Channels c
-                JOIN
-                    ChannelMembers cm on c.Id = cm.ChannelId
-                JOIN
-                    Users u on u.Id = cm.UserId
-                WHERE
-                    c.Type = :ChannelType
-                AND
-                    u.Id = :UserId
-                GROUP BY
-                    c.Id
-            ) cc
-            JOIN
-                ChannelMembers cm on cc.Id = cm.ChannelId
-            JOIN
-                Users u on u.Id = cm.UserId
-            GROUP BY
-                cc.Id
-            HAVING
-                %s
-            LIMIT
-                ` + strconv.Itoa(model.ChannelSearchDefaultLimit)
-	}
-
-	var likeClauses []string
-	args := map[string]interface{}{"UserId": userId, "ChannelType": model.ChannelTypeGroup}
+func (s SqlChannelStore) searchGroupChannelsQuery(userId, term string, isPostgreSQL bool) sq.SelectBuilder {
+	var baseLikeTerm string
+
 	terms := strings.Split(strings.ToLower(strings.Trim(term, " ")), " ")
 
-	for idx, term := range terms {
-		argName := fmt.Sprintf("Term%v", idx)
+	having := sq.And{}
+
+	if isPostgreSQL {
+		baseLikeTerm = "ARRAY_TO_STRING(ARRAY_AGG(u.Username), ', ') LIKE ?"
+		cc := sq.Select("c.Id").
+			From("Channels c").
+			Join("ChannelMembers cm ON c.Id=cm.ChannelId").
+			Join("Users u on u.Id = cm.UserId").
+			Where(sq.And{
+				sq.Eq{"c.Type": model.ChannelTypeGroup},
+				sq.Eq{"u.id": userId},
+			}).
+			GroupBy("c.Id")
+
+		for _, term := range terms {
+			term = sanitizeSearchTerm(term, "\\")
+			having = append(having, sq.Expr(baseLikeTerm, "%"+term+"%"))
+		}
+
+		subq := sq.Select("cc.id").
+			FromSelect(cc, "cc").
+			Join("ChannelMembers cm On cc.Id = cm.ChannelId").
+			Join("Users u On u.Id = cm.UserId").
+			GroupBy("cc.Id").
+			Having(having).
+			Limit(model.ChannelSearchDefaultLimit)
+
+		return s.getQueryBuilder().Select("*").
+			From("Channels").
+			Where(sq.Expr("Id IN (?)", subq))
+
+	}
+
+	baseLikeTerm = "GROUP_CONCAT(u.Username SEPARATOR ', ') LIKE ?"
+
+	for _, term := range terms {
 		term = sanitizeSearchTerm(term, "\\")
-		likeClauses = append(likeClauses, fmt.Sprintf(baseLikeClause, ":"+argName))
-		args[argName] = "%" + term + "%"
+		having = append(having, sq.Expr(baseLikeTerm, "%"+term+"%"))
 	}
 
-	query = fmt.Sprintf(query, strings.Join(likeClauses, " AND "))
-	return query, args
+	cc := sq.Select("c.*").
+		From("Channels c").
+		Join("ChannelMembers cm ON c.Id=cm.ChannelId").
+		Join("Users u on u.Id = cm.UserId").
+		Where(sq.And{
```

Comment:
Just use `sq.Eq` here. You can specify multiple pairs which will be `AND`ed automatically.